### PR TITLE
Custom storage paths

### PR DESCRIPTION
**Pull request by maxwelllevin**
This PR adds support for customizing the output directory structure via the storage configuration file without any breaking changes.

Tasks:
- [X] Implement configuration file options for data & ancillary paths
- [X] Implement variable substitution logic for config options
- [x] Update the `FileSystem.fetch_data()` method to use the custom storage paths
- [x] Add tests to confirm that storage classes are fully compatible with these changes
- [x] Add documentation on allowable variable substitutions

Co-authored-by: James McVey <53623232+jmcvey3@users.noreply.github.com>

### CODE REVIEW

**Comment by maxwelllevin** [comment]
Tests pass locally, the error just seems to be the `metpy` import error from `act-atmos`. I have another PR which excludes the specific offending versions of ACT; v1.3.1 and v1.3.3

**Comment by maxwelllevin** [comment]
Documentation for this feature is primarily in the API reference section. We are planning on making additional changes to support custom file-naming conventions in support of a2e; that will require a slight re-work of some of this code & documentation anyways, so I think just API docs is OK for now.

File: test/config/yaml/storage.yaml [hunk 1]
```diff
@@ -1,4 +1,10 @@
 classname: tsdat.io.storage.FileSystem
+# parameters:
+#   # All these parameters are optional and the values shown below are the defaults
+#   data_folder: data
+#   ancillary_folder: ancillary
+#   data_storage_path: "{storage_root}/{data_folder}/{datastream}"
```
Comment (jmcvey3) [inline test/config/yaml/storage.yaml:6]: We should explicitly show `#  storage_root:  storage/root`

**Comment by maxwelllevin** [comment]
Having trouble with the zarr stuff

**Comment by maxwelllevin** [comment]
Added custom storage paths to the zarr storage class by making `ZarrLocalStorage` inherit from `FileSystem` storage and modifying a few methods as needed

**Pull request merged**